tests/test_totem.lua

```diff
@@ -180,4 +180,28 @@ function tests.test_TensorArgumentErrorMessages()
 end
 
 
+function tests.test_checkGradientsAcceptsGenericOutput()
+  require 'nn'
+  local Mod = torch.class('totem.dummyClass', 'nn.Module')
+  function Mod:updateOutput(input)
+    local ret = {
+        [1] = {
+            [1] = torch.randn(3, 5),
+            [2] = 1,
+            strKey = 3,
+        },
+        [2] = 1,
+        [3] = torch.randn(3, 5),
+        strKey = 4
+    }
+    return ret
+  end
+  function Mod:updateGradInput(input, gradOutput)
+    return input:clone():fill(0)
```

Comment:
you need to do 
```
self.gradInput = input:clone():fill(0)
return self.gradInput
```

otherwise the test will fail. The totem tester builds up a nn.Sequential with a copy layer internally and relies on the module to function properly, in particular to have a gradOutput. 
Also, probably updateOutput should set self.output, though tests would pass without that